fsm.go

```diff
@@ -42,6 +43,9 @@ type transitionerStruct struct {
 //
 // It has to be created with NewFSM to function properly.
 type FSM struct {
+	// mutex is used for synchronization to achieve thread safety
+	mutex sync.RWMutex
```

Comment:
If it was anonymous, wouldn't the Lock/RLock/Unlock/RUnlock methods be public? Thus external access would be dangerous.